plugins/woocommerce/client/admin/client/products/product-page.tsx

```diff
@@ -71,9 +71,11 @@ export default function ProductPage() {
 				return (
 					<>
 						<WooProductMoreMenuItem>
-							{ ( { onClose }: { onClose: () => void } ) => (
-								<MoreMenuFill onClose={ onClose } />
-							) }
+							{
+								( ( { onClose }: { onClose: () => void } ) => (
+									<MoreMenuFill onClose={ onClose } />
+								) ) as unknown as React.ReactNode
+							}
```

Comment:
I know it wasn't introduced in this PR, but I'm a bit confused by this pattern. :thinking: Why are we defining a function that takes the `onClose` prop simply to pass it down to a component? And the usage of  `as unknown` seems not ideal IMO.

I wonder if this could be refactored to simply:

```suggestion
							<MoreMenuFill />
```

And then make the `onClose` prop in `MoreMenuFill` optional. :thinking:

Reply:
This is a Slot Fill pattern used in multiple places in Woo. I'm not convinced to that either.

> I wonder if this could be refactored to simply:

The internal logic behaves differently when you provide React component vs function returning component. So it should not be the way to go.

> And the usage of as unknown seems not ideal IMO.

But I agree with that 😅 let me try to adjust the internal types to accommodate the case with function returning a component as such.

Reply:
I spent some time investigating it but with no luck. I even consulted this with @gigitux and we cannot figure it out. I added `ts-expect-error` for now in those two cases and if we're ok with that, I'll create a dedicated issue for these two. I know it's not ideal.

Reply:
@louwie17, thank you ❤️ This one and the other similar is fixed in https://github.com/woocommerce/woocommerce/pull/54129/commits/bcd77ab3f81f79d319206f930231bfa7525288f1

Reply:
Thanks for the explanation, @louwie17! It makes sense.